Libraries/LibWeb/HTML/Numbers.cpp

```diff
@@ -90,15 +100,26 @@ Optional<StringView> parse_non_negative_integer_digits(StringView string)
 }
 
 // https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#rules-for-parsing-non-negative-integers
-Optional<u32> parse_non_negative_integer(StringView string)
+Optional<u32> parse_non_negative_integer(StringView string, Optional<u32> minimum, Optional<u32> maximum)
 {
     auto optional_digits = parse_non_negative_integer_digits(string);
     if (!optional_digits.has_value())
         return {};
 
     auto optional_value = optional_digits->to_number<i64>(TrimWhitespace::No);
-    if (!optional_value.has_value() || *optional_value > NumericLimits<u32>::max())
+    if (!optional_value.has_value()) {
+        if (maximum.has_value())
+            return maximum;
+
         return {};
+    }
+
+    auto value = optional_value.value();
+    if (minimum.has_value() && value < minimum.value())
+        return minimum;
+
+    if (maximum.has_value() && value > maximum.value())
+        return maximum;
 
     return static_cast<u32>(optional_value.value());
```

Comment:
Because `maximum` is optional and you've removed the `*optional_value > NumericLimits<u32>::max()` check, we could now try to cast a number that is larger than `NumericLimits<u32>::max()`.